frontend/institution/allInstitutionsController.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+(function() {
+    var app = angular.module('app');
+
+    app.controller("AllInstitutionsController", function AllInstitutionsController(
+        $state, InstitutionService, AuthService, MessageService) {
+        var allInstituinsCtrl = this;
+        allInstituinsCtrl.isLoadingInstitutions = true;
+
+        allInstituinsCtrl.user = AuthService.getCurrentUser();
+
+        function loadInstitutions() {
+            InstitutionService.getInstitutions().then(function success(response) {
+                allInstituinsCtrl.institutions = response.data;
+                allInstituinsCtrl.isLoadingInstitutions = false;
+            }, function error(response) {
+                $state.go('app.user.home');
+                MessageService.showToast(response.data.msg);
+            });
+        }
+
+        loadInstitutions();
+    });
+
+    app.controller("InstitutionCardController", function InstitutionCardController(
+        $state, AuthService, InstitutionService, MessageService, ngClipboard){
+        var institutionCardCtrl = this;
+        var URL_INSTITUTION = '/institution/';
+
+        institutionCardCtrl.user = AuthService.getCurrentUser();
+
+        institutionCardCtrl.goToInstitution = function goToInstitution(institutionKey) {
+            $state.go('app.institution.timeline', {institutionKey: institutionKey});
+        };
+
+        institutionCardCtrl.generateLink = function generateLink(){
+            var currentUrl = (window.location.host);
+            var url = currentUrl + URL_INSTITUTION + institutionCardCtrl.institution.key + "/home";
+            ngClipboard.toClipboard(url);
+            MessageService.showToast("O link foi copiado");
+        };
+
+        institutionCardCtrl.showFollowButton = function showFollowButton() {
```

Comment:
That logic it's being spread all over the code, it must be centralized to avoid bugs and facilitate future changes.